src/tsd.c

```diff
@@ -12,6 +12,40 @@ malloc_tsd_data(, , tsd_t, TSD_INITIALIZER)
 
 /******************************************************************************/
 
+void
+tsd_slow_update(tsd_t *tsd) {
```

Comment:
This diff hasn't added the reentrancy behind the tsd_slow flag yet. The corresponding change is in the latter diff.